### PR TITLE
Use kibiUtils.getValuesAtPath in favor of _.get

### PR DESCRIPTION
**Pull request by RubieV**
This adds the feature to get arrays from the es result document, by reusing kibiUtils.getValuesAtPath with .split('.'). This feature might be implemented on a higher level of abstraction, to get rid of params.labelFieldSequence statements altogether.

### CODE REVIEW

**Comment by szydan** [comment]
jenkins test it

**Comment by RubieV** [comment]
Failing: Kibi Timeline Kibi Timeline TimelineHelper pluckLabel should return the label of an event kibana-style
Will update tests to make it pass.

**Comment by scampi** [comment]
Thanks for the patch. The parameter `labelFieldSequence` will not be removed because it is used for supporting dots in field names, although it is a controversial feature.

**Comment by RubieV** [comment]
@scampi I prefer to keep `labelFieldSequence`, and move the Kibana's labelField to also make use of `labelFieldSequence`. It seems that the existence of the `fields` marks the difference between Kibi and Kibana. When confirmed, we can move to only use labelFieldSequence uniformly.

Initialisation:
https://github.com/sirensolutions/kibi_timeline_vis/blob/220ad1747558774d32b0a2d9f982b7e887bfe18a/public/kibi_timeline_vis_controller.js#L80-L86

Usage:
https://github.com/sirensolutions/kibi_timeline_vis/blob/4545908f8e5afc09cd4fd33f16defcf891eca54e/public/lib/helpers/timeline_helper.js#L28-L32

**Comment by RubieV** [comment]
781b57302f277e5abf75b9446fd0fa58ce8f198d fixes case where there is not split on dot, forces array creation.

**Comment by scampi** [comment]
> This feature might be implemented on a higher level of abstraction, to get rid of params.labelFieldSequence statements altogether.

You meant Kibana ? If so, sorry I misread this. In kibana the support for objects in arrays has been improved and so it shouldn't be needed to use the `kibiUtils` version. However, in Kibi we support dots in fieldnames which kibana will not. If we choose to still support this, then those statements are still needed.

**Comment by scampi** [comment]
jenkins test it

**Comment by RubieV** [comment]
Sorry for the late commits, I've updated the build artifact and added test coverage for Kibana. I had issues running the tests, so please let Jenkins do a check.

**Comment by scampi** [comment]
jenkins test it

File: public/lib/helpers/timeline_helper.js [hunk 1]
```diff
@@ -27,8 +27,10 @@ define(function (require) {
       let field;
       if (params.labelFieldSequence) { // in kibi, we have the path property of a field
         field = kibiUtils.getValuesAtPath(hit._source, params.labelFieldSequence);
-      } else {
-        field = _.get(hit._source, params.labelField);
+      } else { // create kibi path property from ES path by splitting on .
+        const kibanaSplit = params.labelField.split('.');
+        const kibanaSequence = Array.isArray(kibanaSplit) ? kibanaSplit : Array.of(kibanaSplit);
```
Comment (scampi) [inline public/lib/helpers/timeline_helper.js]: I don't understand why this is needed. The `split` always returns an array.

**Comment by szydan** [comment]
jenkins test it

**Comment by scampi** [comment]
@RubieV please accept the PR https://github.com/RubieV/kibi_timeline_vis/pull/1

**Comment by RubieV** [comment]
Accepted

**Comment by szydan** [comment]
jenkins test it

**Comment by RubieV** [comment]
Sorry to bother, any updates on the status?

**Comment by scampi** [comment]
@RubieV finishing this in https://github.com/sirensolutions/kibi_timeline_vis/pull/123 thanks

**Pull request merged**